### PR TITLE
Updated Function96 with V2 files and removed chconf.h and halconf.h

### PR DESCRIPTION
**Pull request by MatthieuLu**
Updated the Function96 with V2 files and separated the revisions.

## Description

 Updated the Function96 with V2 files and separated the revisions. This includes all the layouts supported. Also removed the chconf.h and halconf.h files as they seem to be unnecessary for both V1 and V2.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/function96/v1/mcuconf.h [hunk 1]
```diff
@@ -0,0 +1,168 @@
+/*
```
Comment (drashna) [inline keyboards/function96/v1/mcuconf.h:1]: Please run `qmk chibios-confmigrate` on this file.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/function96/v1/mcuconf.h (hunk 1 above)
Comment (MatthieuLu) [reply keyboards/function96/v1/mcuconf.h:1]: Sounds good: What should I input as the -r (Reference) and -i (Input)?

**Review by drashna — APPROVED** [review]

File: keyboards/function96/v2/config.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+#include "config_common.h"
+
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x2A34 // JT , JTMDesign
+#define PRODUCT_ID   0x672B
+#define DEVICE_VER   0x0002
+#define MANUFACTURER JTMDesign
+#define PRODUCT      JTMDesign function96v2
```
Comment (fauxpark) [inline keyboards/function96/v2/config.h:28]:
```suggestion
#define PRODUCT      function96v2
```

File: keyboards/function96/v2/keymaps/ansi_splitspace/keymap.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      )
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96/v2/keymaps/ansi_splitspace/keymap.c:46]:
```suggestion
```

File: keyboards/function96/v2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        _______, _______, _______,_______,_______,_______,_______,_______,_______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,_______,_______,_______,_______,_______,_______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,_______,_______,_______,_______,_______,_______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,_______,_______,_______,_______,_______,_______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,_______,_______,_______,_______,_______,_______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                _______,                         _______, _______, _______, _______, _______, _______, _______, _______
+      ),
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96/v2/keymaps/default/keymap.c:38]:
```suggestion
```

File: keyboards/function96/v2/keymaps/iso/keymap.c [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96/v2/keymaps/iso/keymap.c:38]:
```suggestion
```

File: keyboards/function96/v2/keymaps/iso_splitspace/keymap.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96/v2/keymaps/iso_splitspace/keymap.c:46]:
```suggestion
```

File: keyboards/function96/v2/readme.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Function96v2
+
+This Function96v2 is a passion project of JTM. It's meant to give the user the maximum usability of the space allocated by the keyboard while also giving the option of using the numpad.
+
+* Keyboard Maintainer: The QMK Community
```
Comment (fauxpark) [inline keyboards/function96/v2/readme.md:5]:
```suggestion
* Keyboard Maintainer: The QMK Community
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by Erovia — APPROVED** [review]

**Pull request merged**